README.md

```diff
@@ -12,144 +12,361 @@
-    - A function that creates a timer function based on a specified duration.
-
-14. **Endpoints:**
-    - Statuses in the workflow that do not have any outgoing transitions, indicating the end points of the workflow.
-
-15. **EventStreamer:**
-    - An interface representing a stream for workflow events. It includes methods for producing and consuming events.
-
-16. **Graph:**
-    - A representation of the workflow's structure, showing the relationships between different statuses and transitions.
-
-17. **InternalState:**
-    - A map holding the state of all expected consumers and timeout goroutines using their role names as keys. It is protected by a mutex to ensure thread safety.
-
-18. **MermaidDiagram:**
-    - A function generating a Mermaid diagram for the workflow structure based on the provided Workflow, path, and MermaidDirection.
-
-19. **MermaidDirection:**
-    - A type representing the direction of the Mermaid diagram, such as TopToBottom, LeftToRight, RightToLeft, or BottomToTop.
-
-20. **Not:**
-    - A function that negates the result of another consumer function, used as a filter for steps.
-
-21. **Producer:**
-    - A component that produces events to an event stream. It is responsible for sending events to the stream.
-
-22. **RecordStore:**
-    - An interface representing a store for workflow records. It includes methods for storing and retrieving records.
-
-23. **RoleScheduler:**
-    - An interface representing a scheduler for roles in the workflow. It is responsible for coordinating the execution of different roles.
-
-24. **Run:**
-    - A method in the workflow struct that starts background processes necessary for running the workflow, such as consumers, timeouts, and connectors.
-
-25. **Schedule:**
-    - A method in the workflow API that schedules workflow triggers at specified intervals using a cron-like specification.
-
-26. **State:**
-    - An enumeration representing the state of a consumer or timeout goroutine. Possible states include StateUnknown, StateShutdown, and others.
-
-27. **StepOption:**
-    - A functional option for configuring step-specific settings, such as parallel count, polling frequency, and error backoff.
-
-28. **TimeoutFunc:**
-    - A function type representing a timeout action in the workflow, executed when a specified time duration has elapsed.
-
-29. **TimeoutOption:**
-    - A functional option for configuring timeout-specific settings, such as polling frequency and error backoff.
-
-30. **Topic:**
-    - A method that generates a topic for producing events in the event streamer based on the workflow name and status.
-
-31. **TimerFunc:**
-    - A function type used to calculate the time for a timeout. It takes a context, record, and the current time as inputs.
-
-32. **Trigger:**
-    - A method in the workflow API that initiates a workflow for a specified foreignID and starting status. It returns a runID and allows for additional configuration options.
+**Builder:**
```

Comment:
This is quite a lot to take in! Would adding it to a table make it a little more digestible, and easier to lookup the key?